core/src/test/java/com/pholser/junit/quickcheck/ShrinkingTest.java

```diff
@@ -204,17 +208,17 @@ public void shouldHold(Foo f) {
         assertThat(
             testResult(ShrinkingMoreThanOnePropertyParameter.class),
             hasSingleFailureContaining(
-                String.format("Shrunken args: [%s, %s]", new Foo(1), new Foo(1))));
+                String.format("Shrunken args: [%s, %s]", new Foo(0), new Foo(1))));
```

Comment:
Not sure I understand this - why are they different now?

Reply:
Yeah - come to think of it, why would `0` get admitted here, given the constraints on the parameters? I'll investigate further.

Reply:
I guess there's only a guarantee that generated values will conform to the constraints, but not necessarily shrunken values. Maybe that's OK? Thinking about it some more...

Reply:
In the pathological case, that could cause some issues. Consider a function that is not defined for 0 (that's why the author put the constraint), and a generator that always returns 0 as a shrink. In every failure case, shrinks will be useless.

Reply:
Right. I think it should be up to the generator to ensure that any values it shrinks still conform to any constraints implied by the generator configuration annotations.

Reply:
Ah you're right, I didn't consider that. Doesn't fix the issue with @When constraints, but it's better than I thought.

Reply:
Good point re: `@When(satisfies = ...)`. I'd like to come up with a solution for that, but perhaps down the road...